### PR TITLE
[FIX] web: Fix column_invisible issue

### PR DESCRIPTION
**Pull request by sja-odoo**
Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/web/static/src/js/fields/relational_fields.js [hunk 1]
```diff
@@ -783,8 +783,8 @@ var FieldX2Many = AbstractField.extend({
             return this._super();
         }
         if (this.renderer) {
-            this.renderer.updateState(this.value, {});
             this.currentColInvisibleFields = this._evalColumnInvisibleFields();
+            this.renderer.updateState(this.value, {'columnInvisibleFields': this.currentColInvisibleFields});
             this.pager.updateState({
                 size: this.value.count,
                 columnInvisibleFields: this.currentColInvisibleFields,
```
Comment (ged-odoo) [inline addons/web/static/src/js/fields/relational_fields.js]: you can also remove this line.  This is actually coming from the original commit, and it is wrong.  The columnInvisibleFields was supposed to be given to the renderer, not the pager

File: addons/web/static/tests/fields/relational_fields_tests.js [hunk 2]
```diff
@@ -9475,6 +9478,13 @@ QUnit.module('relational_fields', {
         assert.strictEqual(form.$('th').length, 2,
             "should be 2 columns in the one2many");
         form.$buttons.find('.o_form_button_edit').click();
+        form.$('.o_field_many2one[name="product_id"] input').click();
+        $('li.ui-menu-item a:contains(xpad)').trigger('mouseenter').click();
+        assert.strictEqual(form.$('th').length, 1,
+            "should be 1 column when the product_id is set");
+        $('.o_field_many2one[name="product_id"] input').val('').trigger('keyup');
```
Comment (ged-odoo) [inline addons/web/static/tests/fields/relational_fields_tests.js]: do not forget the form. before $: $(...) is a global selector, and form.$(...) is a selector restricted to the DOM of the form widget

**Comment by sja-odoo** [comment]
@ged-odoo changes done

**Comment by wtaferner** [comment]
@ged-odoo @jco-odoo @pimodoo @sle-odoo @amoyaux 
Not sure what this is trying to fix heire but I have an issue with the column_invisible attribute in 11.0 in the ```stock.picking``` form view in combination with ```stock.production.lot``` when you use the sort mechanism or in case you add a new line of a x2m field.

I guess I need to open an issue at some point, but if it is a known issue I can spare this and maybe one of you has stumbled across this already...little feedback would help and I open immediately a ticket for you guys to fix it ;-)

**Pull request merged**